### ISSUE TITLE
Eobs fails when no channels of an instrument are used (iuse=0)

### ISSUE THREAD

**Issue by CatherineThomas-NOAA** [NOAA-EMC/GSI#133]
@DavidHuber-NOAA found an interesting bug in the GSI observer.  He set all channels for SAPHIR to iuse=0 because of restricted data issues.  Eobs fails with the following message:
> **   ERROR: Can't write definitions - NetCDF4 layer not initialized yet!
> ** Failed to process data/write NetCDF4.

For most of the netCDF diag file generation in setuprad, there are IF blocks to prevent the writing of diag files if no channels are used, such as lines 755-758:
`  if (rad_diagsave .and. nchanl_diag > 0) then`
`     if (binary_diag) call init_binary_diag_`
`   if (netcdf_diag) call init_netcdf_diag_`
`endif`

However, there is a block that does not include a check for nchanl_diag (Line 2111):
`  if (rad_diagsave) then`
`     if (netcdf_diag) call nc_diag_write`
`     if (binary_diag) call final_binary_diag_`
`     if (lextra .and. allocated(diagbufex)) deallocate(diagbufex)`
`  endif`

Dave tested adding  `nchanl_diag>0` to that IF block and eobs passed.  This seems to me to be a needed fix but I'd appreciate someone with more setuprad experience take a look. @emilyhcliu @HaixiaLiu-NOAA

**Comment by emilyhcliu** [comment]
@DavidHuber-NOAA @CatherineThomas-NOAA @HaixiaLiu-NOAA 

For eobs step, the reduce_diag is set to true.  And if iuse is also set to zero, then nchanl_diag will be zero.  (see the following code block)

```
!  Find number of channels written to diag file
  if(reduce_diag)then
     nchanl_diag=0
     do i=1,nchanl
        if(iuse_rad(ich(i)) >= 1)then
           nchanl_diag=nchanl_diag+1
           ich_diag(nchanl_diag)=i
        end if
     end do
     if(mype == mype_diaghdr(is))write(6,*)'SETUPRAD:  reduced number of channels ',&
        nchanl_diag,' of ',nchanl,' written to diag file '
  else
     nchanl_diag=nchanl
     do i=1,nchanl_diag
        ich_diag(i)=i
     end do
  end if

```
The failure is due to trying to write to a NetCDF file that is not initialized at all.  
So, I think if we set SAPHIR to monitoring only (iuse=0), then we do not need a diagnostic file for eobs step, then the fix from David makes sense.  The fix prevents writing to a NetCDF file that is not initialized.

**Comment by MichaelLueken-NOAA** [comment]
These changes have been merged to the authoritative repo at 51a6564. Closing this issue.
